scalasql/core/src/DbApi.scala

```diff
@@ -187,9 +228,16 @@ object DbApi {
       connection: java.sql.Connection,
       config: Config,
       dialect: DialectConfig,
-      autoCommit: Boolean,
-      rollBack0: () => Unit
```

Comment:
I've removed the `rollBack0` argument since it was only used to handle the difference between `autoCommit` true/false behavior.

With the new interface, client code can now hook into the rollback process if needed.